### PR TITLE
Fix vertical space before equation in latex (closes: #4574)

### PR DESCRIPTION
**Pull request by jfbu**
Relates #4574

### CODE REVIEW

**Review by tk0miya — APPROVED** [review]
LGTM

File: sphinx/writers/latex.py [hunk 1]
```diff
@@ -1959,6 +1959,10 @@ def add_target(id):
             if id.startswith('index-'):
                 return
 
+            # equations also need no extra blank line nor hypertarget
+            if id.startswith('equation-'):
```
Comment (tk0miya) [inline sphinx/writers/latex.py]: math elements are provided as extension. According to basis, LaTeX writer should not know about `equation-` label.
Could you add a TODO here please?

**Comment by jfbu** [comment]
Thanks for review @tk0miya. I had originally

    math_eqref_format = 'Equation number {number}'

in my test file to double-check that the change did not introduce conflict with that feature. But at last minute I did not include that in the test file of the PR.

(I was surprised that the change needed no update to test for `make test` to pass it seems math equations are not much tested)

File: sphinx/writers/latex.py (hunk 1 above)
Comment (jfbu) [reply sphinx/writers/latex.py]: OK, I will. Aren't you planning to merge mathbase into Sphinx-core ?

Comment (tk0miya) [reply sphinx/writers/latex.py]: Yes! see #4333

**Comment by tk0miya** [comment]
LGTM!

**Pull request merged**